### PR TITLE
[Documents] Get Available Sites endpoint

### LINKED ISSUE #592: [Documents] Get Available Sites endpoint
Add endpoint for `/admin/settings/get-available-sites`

Part of https://github.com/pimcore/studio-backend-bundle/issues/566

### PR DESCRIPTION
**Pull request by lukmzig**
## Changes in this pull request
Resolves #592

### CODE REVIEW

File: src/Document/Schema/Site.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+use OpenApi\Attributes\Items;
+use OpenApi\Attributes\Property;
+use OpenApi\Attributes\Schema;
+use Pimcore\Bundle\StudioBackendBundle\Util\Schema\AdditionalAttributesInterface;
+use Pimcore\Bundle\StudioBackendBundle\Util\Trait\AdditionalAttributesTrait;
+
+#[Schema(
+    schema: 'Site',
+    title: 'Site',
+    required: [
```
Comment (martineiber) [inline src/Document/Schema/Site.php:28]: Should we add the required properties? id, domain, domains?

File: src/Document/Service/SiteService.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    public function getAvailableSites(ExcludeMainSiteParameter $parameter): array
+    {
+        $sites = [];
+        if ($parameter->getExcludeMainSite() === false) {
+            $sites[] = $this->getMainSite();
+        }
+
+        $siteList = $this->siteRepository->listSites();
+        foreach ($siteList as $siteEntry) {
+            $site = new Site(
```
Comment (martineiber) [inline src/Document/Service/SiteService.php]: Move this to the an Hydrator?

**Review by martineiber — CHANGES_REQUESTED** [review]

File: src/Document/Schema/Site.php (hunk 1 above)
Comment (mattamon) [reply src/Document/Schema/Site.php:28]: yes we should :)

**Comment by lukmzig** [comment]
@martineiber thanks for the review :) changed are now included

**Review by martineiber — APPROVED** [review]
LGTM :seal:

**Pull request merged**